### PR TITLE
HHH-11647 - Use ALTER TABLE IF EXISTS on Postgres

### PR DESCRIPTION
**Pull request by dreis2211**
Fixes https://hibernate.atlassian.net/browse/HHH-11647

### CODE REVIEW

**Comment by vladmihalcea** [comment]
Thanks for the PR. Check out [my comment on the Jira issue](https://hibernate.atlassian.net/browse/HHH-11647?focusedCommentId=93302&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-93302). We need to figure out wht the `dialect.supportsIfExistsAfterConstraintName` did not handle it, and implement it according to `supportsIfExistsAfterConstraintName` as well.

**Comment by dreis2211** [comment]
Thanks! Also see [my comment](https://hibernate.atlassian.net/browse/HHH-11647?focusedCommentId=93304&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-93304).

File: hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java [hunk 1]
```diff
@@ -1452,6 +1452,20 @@ public String getCreateTableString() {
...
+	 */
+	public String getAlterTableString() {
+		final StringBuilder sb = new StringBuilder( "alter table" );
+		if ( supportsIfExistsAfterAlterTable() ) {
+			sb.append( " if exists" );
+		}
+		return sb.toString();
+	}
+
+	/**
```
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java]: what about passing the table name?

Comment (dreis2211) [inline hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java]: Yeah, why not - will probably reduce the code duplication even further where those statements are built.

**Comment by dreis2211** [comment]
@dreab8 Thanks for reviewing! I included the table name now. Let me know if I should squash the commits given you're okay with the current state.

**Comment by dreab8** [comment]
@dreis2211 Thanks a lot for the PR and yes you can squash the commits.

 If you haven't already done can you please sign the sign the [CLA](https://cla.jboss.org/)?

Thanks

**Comment by dreis2211** [comment]
@dreab8 Squashed. CLA is signed already - did contributions before that.

**Comment by dreab8** [comment]
@dreis2211 Thanks

**Comment by vladmihalcea** [comment]
Applied upstream, thanks

**Pull request closed without merging**